### PR TITLE
refactor: Split true-impact line count by implementation vs. tests

### PR DESCRIPTION
**Pull request by sahil87**
## Meta

| ID | Type | Confidence | Plan | Review |
|----|------|-----------|------|--------|
| 7t5a | refactor | 5.0/5.0 | 13/13 tasks, 0/0 acceptance | ✓ 1 cycle |

**Pipeline**: [intake](https://github.com/sahil87/fab-kit/blob/260602-7t5a-true-impact-test-split/fab/changes/260602-7t5a-true-impact-test-split/intake.md) ✓ → [apply](https://github.com/sahil87/fab-kit/blob/260602-7t5a-true-impact-test-split/fab/changes/260602-7t5a-true-impact-test-split/plan.md) ✓ → review ✓ → hydrate ✓ → ship → review-pr

**Impact**:
  impl:  +193/−66  (net +127)
  tests: +308/−10  (net +298)
  total: +501/−76  (net +425)  ← excludes `fab/`, `docs/`

## Summary

The `true_impact` block reports a single net line-count that conflates implementation code with test code, even though this project treats tests as first-class deliverables (constitution VII, `test-alongside`). Reviewers can't tell disciplined test coverage apart from implementation bloat, and the pressure to hide tests in `true_impact_exclude` grows — destroying the very signal that proves the work is tested. This change attributes the scaffolding-excluded universe to tests vs. implementation, showing both, rather than excluding tests. Reimplements PR #361's design on the current 6-stage pipeline.

## Changes

- New config key `test_paths` — project-defined glob/pathspec list (no kit default; Portability), graceful collapse when absent/empty.
- New `tests` sub-block in the `true_impact` block — extra `git diff --shortstat` pass within the scaffolding-excluded universe.
- Canonical impact math (the shared engine) — `Result.Tests *Pair`, `testPaths` threaded through `Compute`/`ComputeForRepo`, `runShortstat` extended for include pathspecs.
- Three-row rendering — impl/tests/total in `/git-pr`; compact `{impl}i+{tests}t={total}` in `fab change list --show-stats`.
- Correctness guard (negative-impl clamp) — per-component `impl = max(0, total − tests)` at render time, one-line stderr warning on clamp.
- Edge cases — empty `test_paths` collapses to a single total line; empty `true_impact_exclude` degenerates total to raw.
- CLI and YAML surface — `fab impact` emits the `tests` sub-block; statusfile encode/decode + `impactColumn()` updated.
- Scaffold config placeholder — commented-out `test_paths` knob for discoverability.
- Documentation (constitution-mandated) — `_cli-fab.md`, `SPEC-git-pr.md`, and memory (`schemas`, `configuration`).

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/go/fab/internal/impact/impact.go [hunk 1]
```diff
@@ -61,38 +71,66 @@ func Compute(repoDir, base, head string, excludes []string) (Result, error) {
...
+	return Compute(filepath.Dir(fabRoot), base, head, cfg.TrueImpactExclude, cfg.TestPaths)
 }
 
-func runShortstat(repoDir, base, head string, excludes []string) (int, int, error) {
+// runShortstat runs `git diff --shortstat <base>...<head>` with an optional
+// pathspec built from includes and excludes. When includes is non-empty, each
+// entry is passed as a literal pathspec; otherwise `.` is used as the base
+// path so the `:(exclude)` magic pathspecs apply against the whole tree. Each
+// exclude is appended as a `:(exclude)<pattern>` magic pathspec. When both
+// slices are empty, no `--` pathspec separator is emitted (the raw pass).
```
[the remark this answers is not in the document]
Comment (sahil87) [reply src/go/fab/internal/impact/impact.go]: Fixed — corrected the runShortstat doc comment to state include patterns are passed as `:(glob)` magic pathspecs (matching the implementation), not literal pathspecs. (4d75acb4)

File: src/go/fab/internal/statusfile/statusfile_test.go [hunk 2]
```diff
@@ -355,6 +355,114 @@ last_updated: "2026-04-23T05:02:32Z"
...
+    deleted: 0
+    net: 400
+  computed_at: "2026-05-30T00:00:00Z"
+  computed_at_stage: apply
+`
+
+	dir := t.TempDir()
+	path := filepath.Join(dir, ".status.yaml")
+	os.WriteFile(path, []byte(yamlWithTests), 0644)
+
```
[the remark this answers is not in the document]
Comment (sahil87) [reply src/go/fab/internal/statusfile/statusfile_test.go:383]: Fixed — now checks the os.WriteFile error and fails fast via t.Fatal. (4d75acb4)

File: src/go/fab/internal/statusfile/statusfile_test.go [hunk 3]
```diff
@@ -355,6 +355,114 @@ last_updated: "2026-04-23T05:02:32Z"
...
+  added: 50
+  deleted: 5
+  net: 45
+  computed_at: "2026-05-30T00:00:00Z"
+  computed_at_stage: apply
+`
+	dir := t.TempDir()
+	path := filepath.Join(dir, ".status.yaml")
+	os.WriteFile(path, []byte(yamlNoTests), 0644)
+
```
[the remark this answers is not in the document]
Comment (sahil87) [reply src/go/fab/internal/statusfile/statusfile_test.go:448]: Fixed — now checks the os.WriteFile error and fails fast via t.Fatal. (4d75acb4)

File: src/go/fab/internal/statusfile/statusfile_test.go [hunk 4]
```diff
@@ -355,6 +355,114 @@ last_updated: "2026-04-23T05:02:32Z"
...
+	}
+
+	outPath := filepath.Join(dir, ".status-out.yaml")
+	if err := sf.Save(outPath); err != nil {
+		t.Fatalf("Save failed: %v", err)
+	}
+	raw, _ := os.ReadFile(outPath)
+	if strings.Contains(string(raw), "tests:") {
+		t.Errorf("expected no tests: key when Tests is nil, got:\n%s", raw)
+	}
```
[the remark this answers is not in the document]
Comment (sahil87) [reply src/go/fab/internal/statusfile/statusfile_test.go:470]: Fixed — now checks the os.ReadFile error and fails fast with the read error via t.Fatalf. (4d75acb4)

File: src/go/fab/internal/change/change_test.go [hunk 5]
```diff
@@ -490,3 +490,120 @@ last_updated: "2026-03-10T12:00:00Z"
...
+// TestListWithOptions_ShowStatsTestSplit verifies the compact impl/tests/total
+// split in the --show-stats impact column, including the per-component negative
+// clamp when tests over-counts the total.
+func TestListWithOptions_ShowStatsTestSplit(t *testing.T) {
+	fabRoot := setupChangeFixture(t)
+
+	mkChange := func(id, suffix, trueImpact string) string {
+		folder := "260310-" + id + "-" + suffix
+		dir := filepath.Join(fabRoot, "changes", folder)
+		os.MkdirAll(dir, 0o755)
```
[the remark this answers is not in the document]
Comment (sahil87) [reply src/go/fab/internal/change/change_test.go]: Fixed — mkChange() now checks the os.MkdirAll error and fails fast via t.Fatal. (4d75acb4)

File: src/go/fab/internal/change/change_test.go [hunk 6]
```diff
@@ -490,3 +490,120 @@ last_updated: "2026-03-10T12:00:00Z"
...
+  certain: 0
+  confident: 0
+  tentative: 0
+  unresolved: 0
+  score: 0.0
+stage_metrics: {}
+prs: []
+` + trueImpact + `last_updated: "2026-03-10T12:00:00Z"
+`
+		os.WriteFile(filepath.Join(dir, ".status.yaml"), []byte(y), 0o644)
```
[the remark this answers is not in the document]
Comment (sahil87) [reply src/go/fab/internal/change/change_test.go]: Fixed — mkChange() now checks the os.WriteFile error and fails fast via t.Fatal. (4d75acb4)
